show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt

```diff
@@ -0,0 +1,195 @@
+        }
+        // update the coordinate notations using the initial point
+        toCoordinateNotationFromPoint(initialPoint)
+    }
+
+    /**
+     * Updates the tapped graphic and coordinate notations using the [tappedPoint]
+     */
+    fun onMapTapped(tappedPoint: Point?) {
+        if (tappedPoint != null) {
+            // update the tapped location graphic
+            coordinateLocation.geometry = tappedPoint
+            mapViewState.value.graphicsOverlay.graphics.apply {
+                clear()
+                add(coordinateLocation)
+            }
+            // update the coordinate notations using the tapped point
+            toCoordinateNotationFromPoint(tappedPoint)
+        }
+    }
+
+    /**
+     * Uses CoordinateFormatter to update the UI with coordinate notation strings based on the
+     * given [newLocation] point to convert to coordinate notations
+     */
+    private fun toCoordinateNotationFromPoint(newLocation: Point) {
+        coordinateLocation.geometry = newLocation
+        // use CoordinateFormatter to convert to Latitude Longitude, formatted as Decimal Degrees
+        decimalDegrees.value = CoordinateFormatter.toLatitudeLongitudeOrNull(
+            point = newLocation,
+            format = LatitudeLongitudeFormat.DecimalDegrees,
+            decimalPlaces = 4
+        ) ?: return showErrorDialog()
+
+        // use CoordinateFormatter to convert to Latitude Longitude, formatted as Degrees, Minutes, Seconds
+        degreesMinutesSeconds.value = CoordinateFormatter.toLatitudeLongitudeOrNull(
+            point = newLocation,
+            format = LatitudeLongitudeFormat.DegreesMinutesSeconds,
+            decimalPlaces = 4
+        ) ?: return showErrorDialog()
+
+        // use CoordinateFormatter to convert to Universal Transverse Mercator, using latitudinal bands indicator
+        utm.value = CoordinateFormatter.toUtmOrNull(
+            point = newLocation,
+            utmConversionMode = UtmConversionMode.LatitudeBandIndicators,
+            addSpaces = true
+        ) ?: return showErrorDialog()
+
+        // use CoordinateFormatter to convert to United States National Grid (USNG)
+        usng.value = CoordinateFormatter.toUsngOrNull(
+            point = newLocation,
+            precision = 4,
+            addSpaces = true,
+        ) ?: return showErrorDialog()
+    }
+
+    /**
+     * Uses CoordinateFormatter to update the graphic in the map from the given [coordinateNotation]
+     * string entered by the user. Also calls corresponding method to update all the remaining
+     * [coordinateNotation] strings using the notation [notationType].
+     */
+    fun fromCoordinateNotationToPoint(notationType: NotationType, coordinateNotation: String) {
+        // ignore empty input coordinate notation strings, do not update UI
+        if (coordinateNotation.isEmpty()) return
+        val convertedPoint: Point
+
+        when (notationType) {
```

Comment:
You could probably combine these ie. `val convertedPoint: Point = when (notationType) {....}`